### PR TITLE
Convert to lowercase before generating the collection names or hashes

### PR DESCRIPTION
**Pull request by gtorodelvalle**
- 100% tests passed
- Fixes https://github.com/telefonicaid/IoT-STH/issues/119
- Assigned to @frbattid 

<b>THE `release/0.1.0-using-lowercase-for-collection-names-generation` BRANCH SHOULD NOT BE DELETED UNTIL IT LANDS INTO `develop` (https://github.com/telefonicaid/IoT-STH/pull/123) AND `release_0.1.0` (https://github.com/telefonicaid/IoT-STH/pull/122) TOO :)</b>

### CODE REVIEW

File: src/sth_database.js [hunk 1]
```diff
@@ -326,6 +327,9 @@
     var connection = mongoose.connection.useDb(databaseName);
 
     // Get the connection and notify it via the callback.
+    sthLogger.debug('Getting access to the collection: ' + collectionName, {
```
Comment (frbattid) [inline src/sth_database.js]: Is this log related to @mcarracedo requirement? I can remember him asking for a log printing the accessed collection and database. If it is the case, I think the database name is missing in the log :)

Comment (gtorodelvalle) [inline src/sth_database.js]: Right! Including it ;) Thanks!

Comment (frbattid) [inline src/sth_database.js]: Remember to cherry-pick the commit in all the other PRs

**Comment by gtorodelvalle** [comment]
Fixed in https://github.com/telefonicaid/IoT-STH/commit/2a247f9dcc5cc75d7c0dd84a4ca7d6960bab2e87

**Comment by frbattid** [comment]
LGTM

**Pull request merged**